homeassistant/helpers/config_validation.py

```diff
@@ -732,7 +732,7 @@ def url(value: Any) -> str:
     """Validate an URL."""
     url_in = str(value)
 
-    if urlparse(url_in).scheme in ["http", "https"]:
+    if urlparse(url_in).scheme in ["http", "https", "homeassistant"]:
```

Comment:
This validator is also used for backend checks (e.g. configuration flows), while this feature is only supported by our frontend. 

I think, for that reason, we should not add it here.

../Frenck

Reply:
If so, how could i implement ingress URLs in mqtt config urls then ? Should i create a dedicated cv.url_or_homeassistant_prefix validator ?

https://github.com/home-assistant/core/pull/96100 then ?

Reply:
Made the validator specific. We could also consider making a new generic validator.

Reply:
> If so, how could i implement ingress URLs in mqtt config urls then ? Should i create a dedicated cv.url_or_homeassistant_prefix validator for #96100 ?

This needs to be added to Zigbee2MQTT, I'll contact @Koenkk for this.

Reply:
I already did an MR which got reverted because of HA rejecting it:
https://github.com/fthiery/hassio-zigbee2mqtt/commit/c751cbef87241bd5c9d849b76e234757f1135e69

Reply:
We had actually issues on it (using relative URL's), but it seems it can be readded using the `homeassistant` schema.
I have informed @Koenkk . Feel free to reopen a PR for Zigbee2MQTT.